From 96a1b4a969a5f3c9224d786c79e90d15a47094b0 Mon Sep 17 00:00:00 2001
From: Emeric Brun <ebrun@haproxy.com>
Date: Wed, 16 Dec 2015 15:16:46 +0100
Subject: [PATCH 11/13] BUG/MEDIUM: peers: table entries learned from a remote
 are pushed to others after a random delay.

New sticktable entries learned from a remote peer can be pushed to others after
a random delay because they are not inserted at the right position in the updates
tree.
(cherry picked from commit 234fc3c31e751f8191b9b78fa5fd16663c2627fe)
(cherry picked from commit 8b1a697362977b8392caca3efaf97a5a8a8c782b)
---
 src/peers.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/peers.c b/src/peers.c
index 0564d3d..92b4df0 100644
--- a/src/peers.c
+++ b/src/peers.c
@@ -720,7 +720,7 @@ switchstate:
 							ts = stktable_store(ps->table->table, newts, 0);
 							newts = NULL; /* don't reuse it */
 
-							ts->upd.key= (++ps->table->table->update)+(2^31);
+							ts->upd.key= (++ps->table->table->update)+(2147483648U);
 							eb = eb32_insert(&ps->table->table->updates, &ts->upd);
 							if (eb != &ts->upd) {
 								eb32_delete(eb);
-- 
2.4.10

